apps/sms/js/thread_list_ui.js

```diff
@@ -282,6 +286,71 @@ var ThreadListUI = {
         }
 
         break;
+      case 'contextmenu':
+        if (this.inEditMode) {
+          return;
+        }
+
+        parent = event.target.parentNode;
+        parentThreadId = parent.dataset.threadId;
+        var thread = Threads.get(parentThreadId);
+
+        if (parentThreadId) {
+          event.preventDefault();
+          event.stopPropagation();
+          // Show options per single thread
+          var params = {
+            type: 'action',
+            header: { l10nId: 'thread-options' },
+            items:[]
+          };
+          threadIdToDelete.push(parentThreadId);
+
+          params.items.push(
+            {
+              l10nId: 'delete-thread',
+              method: function deleteThread(threadIdToDelete) {
+                ThreadListUI.performDeletion(threadIdToDelete);
+              },
+              params: [threadIdToDelete]
+            }
+          );
+
+          if(typeof Threads.get(+parentThreadId) !== 'undefined') {
+            var isRead = thread && (thread.unreadCount > 0);
+
+            isRead ? l10nKey = 'mark-as-read' : l10nKey = 'mark-as-unread';
```

Comment:
as said in the other patch, better use `l10nKey = isRead ? 'mark-as-read' : 'mark-as-unread'`